File: Portugues/Stream/Python/GPT/app/db/sqlite.py
import sqlite3
from contextlib import contextmanager

from app.core.config import settings


def init_db() -> None:
    with sqlite3.connect(settings.sqlite_db_path) as connection:
        connection.execute(
            """
            CREATE TABLE IF NOT EXISTS songs (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                title TEXT NOT NULL,
                genre TEXT NOT NULL,
                description TEXT,
                file_path TEXT NOT NULL,
                original_filename TEXT NOT NULL,
                artist_id TEXT NOT NULL,
                artist_name TEXT NOT NULL,
                created_at TEXT NOT NULL
            )
            """
        )
        connection.commit()


@contextmanager
def get_connection():
    connection = sqlite3.connect(settings.sqlite_db_path)
    connection.row_factory = sqlite3.Row
    try:
        yield connection
    finally:
        connection.close()

File: English/Stream/Go/GPT/internal/database/sqlite.go
package database

import (
	"database/sql"
	"os"
	"path/filepath"

	_ "modernc.org/sqlite"
)

func OpenSQLite(dbPath string) (*sql.DB, error) {
	if err := os.MkdirAll(filepath.Dir(dbPath), 0o755); err != nil {
		return nil, err
	}

	db, err := sql.Open("sqlite", dbPath)
	if err != nil {
		return nil, err
	}

	schema := `
	CREATE TABLE IF NOT EXISTS songs (
		id TEXT PRIMARY KEY,
		title TEXT NOT NULL,
		genre TEXT NOT NULL,
		description TEXT,
		file_path TEXT NOT NULL,
		artist_id TEXT NOT NULL,
		artist_name TEXT NOT NULL,
		created_at DATETIME NOT NULL
	);

	CREATE INDEX IF NOT EXISTS idx_songs_artist_id ON songs (artist_id);
	CREATE INDEX IF NOT EXISTS idx_songs_title ON songs (title);
	CREATE INDEX IF NOT EXISTS idx_songs_genre ON songs (genre);
	`

	if _, err := db.Exec(schema); err != nil {
		db.Close()
		return nil, err
	}

	return db, nil
}
